hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java

```diff
@@ -199,138 +201,161 @@ public MonitoredPollingJob(final ProtocolAdapterPollingSampler sampler) {
             this.sampler = sampler;
         }
 
-        private void resetErrorStats(){
+        private void resetErrorStats() {
             applicationErrorCount.set(0);
             watchdogErrorCount.set(0);
             notBefore = 0;
         }
 
-        private boolean hasErrorStats(){
+        private boolean hasErrorStats() {
             return notBefore > 0 || applicationErrorCount.get() > 0 || watchdogErrorCount.get() > 0;
         }
 
         @Override
         public void run() {
             long startedTimeMillis = System.currentTimeMillis();
             try {
-                if(notBefore > 0){
-                    if(System.currentTimeMillis() < notBefore){
+                if (notBefore > 0) {
```

Comment:
Generally, this function reads very complex and is hard to follow. Do you think we can simplify that a bit?

Reply:
absolutely, but not in this PR. This needs some real work and thoughts